### PR TITLE
Add cheat sheets

### PR DESCRIPTION
**Pull request by djacu**
Adds a set of cheat sheets that can be printed and distributed.
Source is LaTeX and SVG files.

4 variants are available as the cartesian product of paper size and colors.
A4 vs Letter paper
Color vs Grayscale

```
nix flake show
git+file:///home/nunya/dev/nixos/marketing?ref=refs/heads/add-cheat-sheet&rev=f979dfda5b730407baa516007e831dbe25e8e369
└───packages
...
    └───x86_64-linux
        ├───cheat-sheet-a4-default: package 'nixos-cheat-sheet'
        ├───cheat-sheet-a4-print: package 'nixos-cheat-sheet'
        ├───cheat-sheet-letter-default: package 'nixos-cheat-sheet'
        └───cheat-sheet-letter-print: package 'nixos-cheat-sheet'
```

A4 paper in color
image

Letter paper in "grayscale/bw"
image

*Note:* There is no official NixOS logo that is monochromatic and dark (only white) so the colored version is used. This isn't a big deal as the primary purpose of the grayscale version is to remove the background color from the boxes to save ink.

### CODE REVIEW

**Comment by djacu** [comment]
@cab404
Wanted to let you know I used the header.svg that you submitted to the artwork repo as the background for the cheat sheet. :)

**Comment by cab404** [comment]
Of course! <3

**Comment by cab404** [comment]
A really nice cheatsheet!

It would probably be nice if grid lined up with logo, and filled the page completely.

**Comment by drupol** [comment]
Here's a quick draft of something I did in the train this morning... took me less than 30 min:

https://typst.app/project/rAN-7MckqNuneSf0955nE4

I couldn't reproduce the nice box borders, I guess it would require a bit more time to do that.

I'm pretty sure that the footprint to build this document using Typst is much smaller than with LaTeX.

**Comment by djacu** [comment]
> Here's a quick draft of something I did in the train this morning... took me less than 30 min:
> 
> https://typst.app/project/rAN-7MckqNuneSf0955nE4
> 
> I couldn't reproduce the nice box borders, I guess it would require a bit more time to do that.
> 
> I'm pretty sure that the footprint to build this document using Typst is much smaller than with LaTeX.

This is already a fantastic start!

**Comment by roberth** [comment]
> purpose of the grayscale version is to remove the background color from the boxes to save ink.

Maybe call it "print" instead of grayscale? Achieving actual full grayscale is easy in print settings; otherwise I think a colored logo is nice.

**Comment by con-f-use** [comment]
I wouldn't bind an attribute set to the symbol `set` in didactic material. Ppl. just learning nix might think it's some magic keyword or something.

File: cheatsheet/main.tex [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+if x > 0
+then 1
+else -1
+    \end{minted}
+\end{ctexpression}
+
+\vfill
+
+\begin{ctexpression}{Attribute Sets}
+{\color{nixdarkblue}Update attribute sets using} //
```
Comment (amozeo) [inline cheatsheet/main.tex:214]:
```suggestion
{\color{nixdarkblue}Union attribute sets using} //
```
Set is not being updated, attributes of these sets are being added together.
Mathematical word from set theory for that is union. I would say it's union with right preference.

Comment (amozeo) [reply cheatsheet/main.tex:214]: I noticed that in nix manual it does say update
you can mark this conversation as resolved

**Review by thilobillerbeck — APPROVED** [review]

**Pull request merged**